ably-ios/ARTEventEmitter.h

```diff
@@ -9,14 +9,102 @@
 #import <Foundation/Foundation.h>
 #import "ARTTypes.h"
 
-@protocol ARTSubscription;
-
 @class ARTRealtime;
 
-@interface ARTEventEmitter : NSObject
+ART_ASSUME_NONNULL_BEGIN
+
+@interface __GENERIC(ARTEventListener, ItemType) : NSObject
+
+- (void)call:(ItemType)argument;
+
+@end
+
+@interface __GENERIC(ARTEventEmitter, EventType, ItemType) : NSObject
+
+- (__GENERIC(ARTEventListener, ItemType) *)on:(EventType)event call:(void (^)(ItemType))cb;
+- (void)on:(EventType)event callListener:(__GENERIC(ARTEventListener, ItemType) *)listener;
 
-- (instancetype)initWithRealtime:(ARTRealtime *)realtime;
-- (id<ARTSubscription>)on:(ARTRealtimeConnectionStateCb)cb;
-- (void)removeEvents;
+- (__GENERIC(ARTEventListener, ItemType) *)onAll:(void (^)(ItemType))cb;
```

Comment:
I thought we agreed on something quite different, see https://github.com/ably/ably-ios/issues/121#issuecomment-169501841.  I thought #158 was an exception because it was an initializer / constructor?